### PR TITLE
Upgrade surefire plugin to 2.22 so jacoco profile will execute correctly

### PR DESCRIPTION
**Pull request by prushforth**
This is a test of appveyor and travis

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x ] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)

The following are required only for core and extension modules, while they are welcomed, but not required, for community modules:
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: src/pom.xml [hunk 1]
```diff
@@ -1514,7 +1514,7 @@
      <printSummary>true</printSummary>
      <testFailureIgnore>${allow.test.failure.ignore}</testFailureIgnore>
      <excludedGroups>${test.excludedGroups}</excludedGroups>
-     <argLine>${argLine} -XX:+IgnoreUnrecognizedVMOptions --illegal-access=debug</argLine>
+     <argLine>@{argLine} -XX:+IgnoreUnrecognizedVMOptions --illegal-access=debug</argLine>
```
Comment (nmco) [inline src/pom.xml]: Probably `$` instead of `@`, Maven complains about it, Maven specific version required?

**Comment by nmco** [comment]
Tried this locally, JaCoCo complains about:
```
[INFO] --- jacoco-maven-plugin:0.8.1:report (post-unit-test) @ gs-status-monitoring ---
[INFO] Skipping JaCoCo execution due to missing execution data file.
```

May be specific to my system, not sure if this was what this PR was aiming to fixing :smile:

File: src/pom.xml (hunk 1 above)
Comment (prushforth) [reply src/pom.xml]: @nmco The [surefire](http://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html) documentation says it resolves problems with `<argLine>`, which was what was happening on my machine. I found that I had to remove the `<argLine>` entirely from the root pom.xml in order to get jacoco to run properly.  

> [INFO] --- jacoco-maven-plugin:0.8.1:report (post-unit-test) @ gs-status-monitoring ---
> [INFO] Skipping JaCoCo execution due to missing execution data file.

Did you try it with the upgraded surefire and the @{argLine} usage?

**Comment by prushforth** [comment]
@nmco 

> not sure if this was what this PR was aiming to fixing

Yes!

**Comment by aaime** [comment]
Tried it out computing the coverage on ``gs-main``, works fine for me, running a Maven 3.6.1, also tried a 3.5.3, don't have older versions.

@nmco wondering if the issue is somehow maven version dependent?

I'd merge the PR if there are no objections.

**Comment by aaime** [comment]
No objections raised, merging.

**Pull request merged**